pkg/services/authn/clients/ext_jwt.go

```diff
@@ -95,8 +90,13 @@ func (s *ExtendedJWT) IsEnabled() bool {
 	return s.cfg.ExtJWTAuth.Enabled
 }
 
-func (s *ExtendedJWT) authenticateAsUser(idTokenClaims,
-	accessTokenClaims *ExtendedJWTClaims) (*authn.Identity, error) {
+func (s *ExtendedJWT) authenticateAsUser(idTokenClaims *authlib.Claims[authlib.IDTokenClaims],
+	accessTokenClaims *authlib.Claims[authlib.AccessTokenClaims]) (*authn.Identity, error) {
+	// Only allow token pairs with matching namespace through
+	if accessTokenClaims.Rest.Namespace != idTokenClaims.Rest.Namespace {
```

Comment:
Does this logic make sense to y'all? It shouldn't happen in practice but this is my failsafe to make sure that services don't send any all powerful tokens here (with `namespace = *`, that is).

Reply:
I think we can start with that and loosen the check if a valid use case comes along.